io-engine/src/grpc/v1/replica.rs

```diff
@@ -106,6 +116,74 @@ impl Default for ReplicaService {
     }
 }
 
+impl From<VolumeSnapshotDescriptor> for ReplicaSnapshot {
+    fn from(s: VolumeSnapshotDescriptor) -> Self {
+        Self {
+            snapshot_uuid: s.snapshot_uuid().to_string(),
+            snapshot_name: s.snapshot_params().name().unwrap_or_default(),
+            snapshot_size: s.snapshot_size(),
+            num_clones: s.num_clones(),
+            timestamp: None, //TODO: Need to update xAttr to track timestamp
+            replica_uuid: s.snapshot_params().parent_id().unwrap_or_default(),
+            replica_size: s.replica_size(),
+            entity_id: s.snapshot_params().entity_id().unwrap_or_default(),
+            txn_id: s.snapshot_params().txn_id().unwrap_or_default(),
+        }
+    }
+}
+
+impl From<Lvol> for ReplicaSnapshot {
+    fn from(snap_lvol: Lvol) -> Self {
+        let blob = snap_lvol.bs_iter_first();
+        let mut snapshot_param: SnapshotParams = Default::default();
+        for attr in SnapshotXattrs::iter() {
+            let mut val = std::ptr::null();
+            let mut size: u64 = 0;
+            let attr_id = attr.name().to_string().into_cstring();
+            let curr_attr_val;
+            unsafe {
+                let _r = spdk_blob_get_xattr_value(
+                    blob,
+                    attr_id.as_ptr(),
+                    &mut val as *mut *const c_void,
+                    &mut size as *mut u64,
+                );
+                curr_attr_val = String::from_raw_parts(
+                    val as *mut u8,
+                    size as usize,
+                    size as usize,
+                );
+            }
+            match attr {
+                SnapshotXattrs::ParentId => {
+                    snapshot_param.set_parent_id(curr_attr_val);
+                }
+                SnapshotXattrs::EntityId => {
+                    snapshot_param.set_entity_id(curr_attr_val);
+                }
+                SnapshotXattrs::TxId => {
+                    snapshot_param.set_txn_id(curr_attr_val);
+                }
+            }
+        }
+        let parent_lvol = Bdev::lookup_by_uuid_str(
+            snapshot_param.parent_id().unwrap().as_str(),
```

Comment:
What if parent_id is None? 
Should we handle that so we don't crash? Then we can use TryFrom instead of From trait

Reply:
Taken a different approach to update the replica uuid and size. Fault case is handled now.